operator/jsonnet/config.libsonnet

```diff
@@ -0,0 +1,198 @@
+local utils = (import 'github.com/grafana/jsonnet-libs/mixin-utils/utils.libsonnet');
+
+{
+  loki: {
+    local withDatasource = function(ds) ds + (
+      if ds.name == 'datasource' then {
+        query: 'prometheus',
+        current: {
+          selected: true,
+          text: 'default',
+          value: 'default',
+        },
+      } else {}
+    ),
+
+    local withNamespace = function(ns) ns + (
+      if ns.label == 'namespace' then {
+        datasource: '${datasource}',
+        current: {
+          selected: false,
+          text: 'openshift-logging',
+          value: 'openshift-logging',
+        },
+        definition: 'label_values(loki_build_info, namespace)',
+        query: 'label_values(loki_build_info, namespace)',
+      } else {}
+    ),
+
+    local defaultLokiTags = function(t)
+      std.uniq(t + ['logging', 'loki-mixin']),
+
+    local replaceMatchers = function(replacements)
+      function(p) p {
+        targets: [
+          t {
+            expr: std.foldl(function(x, rp) std.strReplace(x, rp.from, rp.to), replacements, t.expr),
+          }
+          for t in p.targets
+          if std.objectHas(p, 'targets')
+        ],
+      },
+
+    // dropPanels removes unnecessary panels from the loki dashboards
+    // that are of obsolete usage on our AWS-based deployment environment.
+    local dropPanels = function(panels, dropList, fn)
+      [
+        p
+        for p in panels
+        if !std.member(dropList, p.title) && fn(p)
+      ],
+
+    // mapPanels applies recursively a set of functions over all panels.
+    // Note: A Grafana dashboard panel can include other panels.
+    // Example: Replace job label in expression and add axis units for all panels.
+    local mapPanels = function(funcs, panels)
+      [
+        // Transform the current panel by applying all transformer funcs.
+        // Keep the last version after foldl ends.
+        std.foldl(function(agg, fn) fn(agg), funcs, p) + (
+          // Recursively apply all transformer functions to any
+          // children panels.
+          if std.objectHas(p, 'panels') then {
+            panels: mapPanels(funcs, p.panels),
+          } else {}
+        )
+        for p in panels
+      ],
+
+    // mapTemplateParameters applies a static list of transformer functions to
+    // all dashboard template parameters. The static list includes:
+    // - cluster-prometheus as datasource based on environment.
+    local mapTemplateParameters = function(ls)
+      [
+        std.foldl(function(x, fn) fn(x), [withDatasource, withNamespace], item)
+        for item in ls
+        if item.name != 'cluster'
+      ],
+
+    prometheusRules+: {
+      groups: [
+        g {
+          rules: [
+            r {
+              expr: std.strReplace(r.expr, 'cluster, ', ''),
+              record: std.strReplace(r.record, 'cluster_', ''),
+            }
+            for r in g.rules
```

Comment:
Do we want to drop the rules we do not use anymore? (mostly the ones for the histograms)